pootle/apps/pootle_misc/checks.py

```diff
@@ -1008,13 +1010,17 @@ def get_qualitycheck_schema(path_obj=None):
         Category.COSMETIC: _("Cosmetic"),
         Category.FUNCTIONAL: _("Functional"),
         Category.EXTRACTION: _("Extraction"),
-        Category.NO_CATEGORY: _("Other"),
+        Category.NO_CATEGORY: _("Other")
     }
 
+    cat_id_keys = category_ids.keys()
+    cat_id_values = category_ids.values()
+
     for check, cat in checks.items():
         if cat not in d:
             d[cat] = {
                 'code': cat,
+                'name': cat_id_keys[cat_id_values.index(cat)],
```

Comment:
Isn't it simpler to create an inverse mapping of `category_ids` at the module level and use it right away?
```python
category_codes = {v: k for k, v in category_ids.iteritems()}
...
    'name': category_codes[cat],
...
```

As a side note, we may want to rename `'name'` to `'code'`, and `'code'` to `'id'`; but that's a separate cleanup.